### PR TITLE
User theming

### PR DESCRIPTION
**Pull request by yukimono**
wow
 cool

      wow
screen shot 2015-04-20 at 5 13 31 am
screen shot 2015-04-20 at 5 14 11 am

### CODE REVIEW

**Comment by yukimono** [comment]
It's going to need some tuning before it looks good with the Cosmo theme, but the actual handler and render works and is there.

Edit ``app/views/user/theme.css.scss.erb``

**Comment by pixeldesu** [comment]
Define this please: 
> It's going to need some tuning before it looks good with the Cosmo theme

Is it breaking layout-wise or from the (S)CSS of the new theme changes? I haven't changed that much on the actual layout, except the navigation tabs.

File: lib/user_theme_runtime.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+end
+
+# INFO
+#  app/views/user/theme.css.scss.erb
+#   Literally application.css.scss, just barebones and pure to bootstrap
+#  DB::User_Themes
+#   Add/remove columns per what can be customized, reference in theme.css.scss.erb
+# NOTE
+#  Every user with a theme will REGENERATE an entirely new bootstrap instance for every change
+#  this is stored in /app/assets/stylesheets/userstyle/:user-id/:timestamp.css for caching
```
Comment (nilsding) [inline lib/user_theme_runtime.rb]: Why not something like `/public/styles/:user-id/:timestamp.css`?  Makes more sense there imho.

File: lib/user_theme_runtime.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#  You can do one of two things
+#  In the settings controller compile the theme file when it's generated or
```
Comment (nilsding) [inline lib/user_theme_runtime.rb]: Probably combine it with a new Sidekiq worker

**Comment by yukimono** [comment]
screen shot 2015-04-21 at 1 12 39
screen shot 2015-04-21 at 1 12 50

This is what I mean.

**Comment by yukimono** [comment]
@nilsding 
> Why not something like /public/styles/:user-id/:timestamp.css? Makes more sense there imho.

You can, you'd have to hook into Rails and call the precompiler and compile it from the resulting file, though. I tried running it through Sass::Engine but that just cried over files named ".css.scss", trying to invoke Tilt causes the same issues.

A simpler alternative is to just not recompile bootstrap as a whole and manually find the CSS tags that need their color replaced, but it can be annoying to keep updated.

**Comment by pixeldesu** [comment]
Regarding .css.scss`, these filenames are outdated, but something we did out of habit. The latest Rails version doesn't require extensions to be like this, and we could simply switch to `.scss`or just `.haml` for the templates/partials.

**Comment by yukimono** [comment]
I just did it for consistency, honestly.

**Pull request closed without merging**